### ISSUE TITLE
space:truncate() does not work inside transaction

### ISSUE THREAD

**Issue by cyanide-burnout** [tarantool/tarantool#1871]
space:truncate() does not work inside transaction.

Working behavior:
    space:truncate()
    box.begin()
    for _, tuple in pairs(data) do
      space:replace(tuple)
    end
    box.commit()

Expected behavior:
    box.begin()
    space:truncate()
    for _, tuple in pairs(data) do
      space:replace(tuple)
    end
    box.commit()

**Comment by rtsisyk** [comment]
space:truncate() drops and re-creates all indexes under the hood. As result, space:truncate() is actually DDL operation. Unfortunately, Tarantool doesn't support transactional DDL. This feature is very hard to implement and mostly useless (=there are no real customers for that).

I move this request to wishlist. Let's return to this request again in Tarantool 1.8.

**Comment by GeorgyKirichenko** [comment]
Will be fixed after #3800

**Comment by mkostoevr** [comment]
Since the space truncate does not yield, it's allowed to be called in a transaction now. This had been implemented in #4083.

So the following code works fine in Tarantool 2.11:

```lua
box.cfg{}

s = box.schema.space.create('s')
s:create_index('pk')

for i = 1, 10000 do
    s:insert({i})
end

data = {}
for i = 10001, 20000 do
    table.insert(data, {i})
end

box.begin()
    s:truncate()
    for _, tuple in pairs(data) do
        s:replace(tuple)
    end
    s:truncate()
    for _, tuple in pairs({{1}, {2}}) do
        s:replace(tuple)
    end
box.commit()

print(require('yaml').encode(s:select()))

os.exit()
```

The result will be:

```yaml
---
- [1]
- [2]
...
```
